src/timepicker/timepicker.js

```diff
@@ -55,6 +55,46 @@ angular.module('ui.bootstrap.timepicker', [])
     });
   }
 
+  var min;
+  $scope.$parent.$watch($parse($attrs.min), function(value) {
+    var dt = new Date(value);
+    min = isNaN(dt) ? undefined : dt;
+  });
+
+  var max;
+  $scope.$parent.$watch($parse($attrs.max), function(value) {
+    var dt = new Date(value);
+    max = isNaN(dt) ? undefined : dt;
+  });
+
+  $scope.noIncrementHours = function() {
+    var incrementedSelected = addMinutes( selected, hourStep * 60 );
+    return incrementedSelected > max || (min && incrementedSelected < selected);
```

Comment:
The logic is a bit different than that.  `incrementedSelected > max` is catching the common case, increasing the time would exceed the max.  If max is undefined, `incrementedSelected > max` will be false, which is what we want(to allow the increase).

The other part, `(min && incrementedSelected < selected)` was intended to disable the increment/decrement if the time would rollover(at midnight) if the opposite direction's wouldn't be allowed.  For example, if it is 11:30pm, you cannot increase to 12:30am if the minimum is 1:00am.

This code isn't quite there though.  This just prevents a rollover if the minimum/maximum is defined. For example, if it is 11:30pm, you cannot increase to 12:30am if the minimum is 12:01am.  This should be allowed, as the updated time is still later than the minimum.

I believe the correct code will be `return incrementedSelected > max || (incrementedSelected < selected && incrementedSelected < min);`.  I've manually tested it.  I'll write up some unit tests and submit it.